core-bundle/tests/Twig/Interop/PhpTemplateProxyNodeTest.php

```diff
@@ -33,7 +33,7 @@ public function testCompilesProxyCode(): void
                     function(callable $block) use ($context): string {
                         if ($this->env->isDebug()) { ob_start(); } else { ob_start(static function () { return ''; }); }
                         try { $block($context); return ob_get_contents(); } finally { ob_end_clean(); }
-                    }, $blocks
+                    }, array_filter($blocks, static fn(callable $block): bool => method_exists(self::class, $block[1]))
```

Comment:
Matching via the method name seems a bit strange, can’t we just directly check for the block keys that get passt to the legacy template?

```suggestion
                    }, array_intersect_key($blocks, $this->blocks)
```

This had the same affect in my tests.

Reply:
That was my first attempt as well. Although, I don't recall the reason anymore, I think there were cases where this did not work. :see_no_evil: But let's find out…